core/src/multi_iterator_scanner.rs

```diff
@@ -114,8 +114,8 @@ where
     }
 
     /// Consume the iterator and return the payload.
-    pub fn finalize(self) -> U {
-        self.payload
+    pub fn finalize(self) -> (U, Vec<bool>) {
```

Comment:
nit: perhaps return a `struct` for clarity?

Reply:
done: 76652cba92e3ff63e66c6fe2f75210435995f6e2

I also updated the doc-comment which I missed previously